### PR TITLE
feat: ProjectsList Search, Layout Improvements

### PR DESCRIPTION
**Pull request by khaled-0**
Read commit messages.

Video: https://cdn.discordapp.com/attachments/797576098627977257/1055184881870704670/Screenrecorder-2022-12-22-00-07-13-538.mp4

### CODE REVIEW

File: app/src/main/java/com/besome/sketch/ProjectsFragment.java [hunk 1]
```diff
@@ -213,19 +230,14 @@ private void toExportProjectActivity(int position) {
         startActivity(intent);
     }
 
+    private void showHideSearchBox(boolean hide) {
+       // projectsSearchView.setVisibility(hide ? View.GONE : View.VISIBLE);
```
Comment (JavkhlanK) [inline app/src/main/java/com/besome/sketch/ProjectsFragment.java]: what?

Comment (JavkhlanK) [reply app/src/main/java/com/besome/sketch/ProjectsFragment.java]: my bad, forgot I was looking at a single commit's changes

Comment (khaled-0) [reply app/src/main/java/com/besome/sketch/ProjectsFragment.java]: yea I forgor to commit before adding new stuffs too.
It somehow got left out when deleting

File: app/src/main/java/com/besome/sketch/ProjectsFragment.java [hunk 2]
```diff
@@ -352,10 +375,39 @@ public int a() {
...
+
+            //Show hide based on search query
+            if (matchesQuery(projectMap)) {
+                viewHolder.b.setVisibility(View.VISIBLE);
+                viewHolder.b.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
+            } else {
+                viewHolder.b.setVisibility(View.GONE);
+                viewHolder.b.setLayoutParams(new ViewGroup.LayoutParams(0, 0));
+
+            }
```
Comment (JavkhlanK) [inline app/src/main/java/com/besome/sketch/ProjectsFragment.java:408]: This is a no-go for me. Please make the RecyclerView actually recycle items well.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -2648,4 +2648,6 @@ Please check the URL again."</string>
     <string name="saveToProject" translatable="false">Project</string>
     <string name="selectAll">Select All</string>
     <string name="text_to_search">Text to search</string>
+    <string name="myprojects_list_menu_title_restore_projects">Restore projects</string>
+    <string name="search_projects">Filter by name,package,id</string>
```
Comment (JavkhlanK) [inline app/src/main/res/values/strings.xml:2652]: Please use commas with a space after them, and also `ID` instead of `id`.

**Review by JavkhlanK — CHANGES_REQUESTED** [review]

**Review by JavkhlanK — CHANGES_REQUESTED** [review]
Searching works great with this, but I don't like the search view always taking up space. It would be greater if it was e.g. inside the toolbar, maybe hidden by default unless a search icon is pressed.

The two restore and create new project cards in one row also doesn't seem like a good UI choice imo.

**Comment by JavkhlanK** [comment]
I might work on making the UI better myself, but I'm not sure.

**Comment by indrawanalghifary** [comment]
Add this search features, this great for many projects

**Comment by AbrishPhoenix** [comment]
What about these two same UIs :)

**Comment by JavkhlanK** [comment]
Two FABs is an absolute no-go, and the always-visible search bar is wasting space.

**Comment by khaled-0** [comment]
Sick gamer toolbar

File: app/src/main/res/layout/toolbar.xml [hunk 4]
```diff
@@ -5,21 +5,19 @@
     android:layout_width="match_parent"
     android:layout_height="@dimen/action_bar_height"
     android:background="?colorPrimary"
+    android:contentInsetStart="0dp"
```
Comment (JavkhlanK) [inline app/src/main/res/layout/toolbar.xml:8]: This seems duplicate

File: app/src/main/res/menu/projects_fragment_menu.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    tools:ignore="AppCompatResource">
```
Comment (JavkhlanK) [inline app/src/main/res/menu/projects_fragment_menu.xml:5]: This ignore isn't needed

File: app/src/main/java/com/besome/sketch/MainActivity.java [hunk 6]
```diff
@@ -277,62 +261,10 @@ public void onDestroy() {
...
-        getMenuInflater().inflate(R.menu.main_menu, menu);
-        return super.onCreateOptionsMenu(menu);
-    }
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
-        if (item.getItemId() == R.id.sortProject) {
-            if (!mB.a()) showProjectSortingDialog();
-        }
+        // ActionBarDrawerToggle#onOptionsItemSelected(MenuItem)
```
Comment (JavkhlanK) [inline app/src/main/java/com/besome/sketch/MainActivity.java:267]: This comment seems off-place

File: app/src/main/java/com/besome/sketch/MainActivity.java [hunk 7]
```diff
@@ -382,7 +314,8 @@ private void allFilesAccessCheck() {
                 dialog.a("Skip", Helper.getDialogDismissListener(dialog));
                 dialog.configureDefaultButton("Don't show anymore", v -> {
                     try {
-                        optOutFile.createNewFile();
+                        if (!optOutFile.createNewFile())
+                            throw new IOException("Unknown error.");
```
Comment (JavkhlanK) [inline app/src/main/java/com/besome/sketch/MainActivity.java:318]: Misleading message: The error actually is known

Comment (JavkhlanK) [reply app/src/main/java/com/besome/sketch/MainActivity.java:318]: If you don't feel like specifying the whole reason, better just leave it as-is

**Review by JavkhlanK — CHANGES_REQUESTED** [review]
That search view inside the toolbar looks and feels very nice. Just one thing, I don't think having a FAB that opens a dialog is okay. I'd prefer still having a CardView that handles the restore logic, and the FAB always opening the New Project dialog.

**Comment by JavkhlanK** [comment]
i'll work on some changes to this PR now, and if I get to make it how I'd like it, I'll merge this alongside your commits

**Pull request merged**